src/ucs/config/parser.c

```diff
@@ -1001,11 +1001,24 @@ static void ucs_config_parser_mark_env_var_used(const char *name, int *added)
         goto out;
     }
 
-    kh_put(ucs_config_env_vars, &ucs_config_parser_env_vars, key, &ret);
+#ifndef __clang_analyzer__
+    /* Exclude this code from Clang examination as it generates
+     * false-postive warning about potential leak of memory
+     * pointed to by 'key' variable */
+    iter = kh_put(ucs_config_env_vars, &ucs_config_parser_env_vars, key, &ret);
+    if ((ret <= 0) || (iter == kh_end(&ucs_config_parser_env_vars))) {
+        ucs_warn("kh_put(key=%s) failed", key);
+        ucs_free(key);
+        goto out;
+    }
+#else
+    ucs_free(key);
+#endif
+
     *added = 1;
```

Comment:
I'd move  `added` under `ifndef`

Reply:
I agree, but I think Clang will report that some code is dead (code under `added > 0`)
btw, if Clang won't complain about this, we will have code that won't be checked by Clang, because `added` is always `0`
wdyt?

Reply:
ok